templates/reservation/reservation.py

```diff
@@ -50,8 +65,9 @@ def install(self):
             raise ValueError("unsupported reservation type %s size %s" % (self.data['type'], self.data['size']))
 
         install_result = install(self.data['size'])
+        self.extend(duration)
         self.state.set('actions', 'install', 'ok')
-        return install_result
+        return install_result, self.data["expiryTimestamp"]
```

Comment:
I don't think you can return 2 values from an action.
I think this will create an undefined behavior on the client side that doesn't not how to deal with this value.